platforms/hyperledger-fabric/configuration/roles/create/new_organization/channel_artifacts/tasks/main.yaml

```diff
@@ -0,0 +1,68 @@
+#############################################################################################
+# This role generate config.json for new organization
+#############################################################################################
+
+- name: check configtxgen
+  stat:
+    path: "{{ build_path }}/configtxgen"
+  register: config_stat_result
+
+- name: register temporary directory
+  tempfile:
+    state: directory
+  register: tmp_directory
+
+############################################################################################
+# This task fetches the configtx gen tar file from the mentioned URL
+- name: "Geting the configtxgen binary tar"
+  get_url:
+    url: https://nexus.hyperledger.org/content/repositories/releases/org/hyperledger/fabric/hyperledger-fabric/{{fabric.os}}-{{fabric.arch}}-{{network.version}}/hyperledger-fabric-{{fabric.os}}-{{fabric.arch}}-{{network.version}}.tar.gz
+    dest: "{{ tmp_directory.path }}"
+  when: config_stat_result.stat.exists == False
+
+############################################################################################
+# This task unzips the above downloaded tar file
+- name: "Unziping the downloaded file"
+  unarchive:
+    src: "{{ tmp_directory.path }}/hyperledger-fabric-{{fabric.os}}-{{fabric.arch}}-{{network.version}}.tar.gz"
+    dest: "{{ tmp_directory.path }}"
+  when: config_stat_result.stat.exists == False
+
+############################################################################################
+# This task extracts the configtxgen binary and place it at appropriate path
+- name: "Moving the configtxgen from the extracted folder and place in it path"
+  copy:
+    src: "{{ tmp_directory.path }}/bin/configtxgen"
+    dest: "{{ build_path }}/configtxgen"
+    mode: 0755
+  when: config_stat_result.stat.exists == False
+
+############################################################################################
+# This task creates the channel-artifacts folder
+- name: "Creating channel-artifacts folder"
+  file:
+    path: "{{ build_path }}/channel-artifacts"
+    state: directory
+
+# This task creates the channel by consuming the configtx.yaml file
+- name: "Creating JSON configration for new organization"
+  shell: |
+    cd {{ build_path }}
+    export FABRIC_CFG_PATH=$PWD
+    ./configtxgen -printOrg {{ participant.name }}MSP > ./channel-artifacts/{{ channel_name | lower }}.json
+  loop: "{{ item.participants }}"
+  loop_control:
+    loop_var: participant
+  when: participant.org_status == 'new'
```

Comment:
Same comment as channels_join. This is the only line difference between this file and platforms\hyperledger-fabric\configuration\roles\create\channel_artifacts\tasks\main.yaml . This adds on maintenance. Please modify the create/channel_artifacts role to take into account new org status ie pass the org_status as a variable.